LiteCore/Support/LogDecoder.cc

```diff
@@ -53,33 +57,26 @@ namespace litecore {
 
 
     void LogIterator::writeTimestamp(Timestamp t, ostream &out) {
-        struct tm tm;
-        localtime_r(&t.secs, &tm);
-        char timestamp[100];
-        strftime(timestamp, sizeof(timestamp), "%T", &tm);
-        out << timestamp;
-        sprintf(timestamp, ".%06u| ", t.microsecs);
-        out << timestamp;
+        local_time<microseconds> tp { seconds(t.secs) + microseconds(t.microsecs) };
+        struct tm tmpTime = FromTimestamp(duration_cast<seconds>(tp.time_since_epoch()));
+        tp -= GetLocalTZOffset(&tmpTime, true);
+        out << format("%T| ", tp);
```

Comment:
Do I understand correctly that the original code writes time in local timezone but the new code writes time in UTC?

Reply:
There should be no changes to the format.  timestamps are always in UTC, so to get the local time the offset needs to be applied. Despite the name "local_time" it doesn't mean "time in the local timezone".  It actually means "a time with no associated time zone"  Contrast this with "sys_time" which is always in UTC.  This plays out more when parsing strings with offsets (sys_time will reverse the offset to bring it to UTC while local time will just ignore the offset).